opencl-aot/source/main.cpp

```diff
@@ -426,9 +556,30 @@ int main(int Argc, char *Argv[]) {
   }
   std::cout << "Using build options: " << BuildOptions << '\n';
 
-  // step 8: build OpenCL program
-  CLErr = clBuildProgram(ProgramUPtr.get(), 1, &DeviceId, BuildOptions.c_str(),
-                         nullptr, nullptr);
+  // step 8: compile | build | link OpenCL program
+  if (OptCommand.getNumOccurrences() && OptCommand == Commands::compile) {
+    assert(OptDevice == fpga_fast_emu &&
+           "Use -cmd=compile with --device=fpga_fast_emu only");
```

Comment:
Why do we have this assertion? `clCompileProgram` is a pretty standard API which must be supported regardless of target device (at least in OpenCL 1.2, 2.*)

Reply:
Removed.